uber/site_sections/registration.py

```diff
@@ -774,16 +774,18 @@ def reg_take_report(self, session, **params):
                                                        Attendee.registered > start, Attendee.registered <= end).all()
             params['sales'] = sales
             params['attendees'] = attendees
-            params['total_cash'] = sum(a.amount_paid for a in attendees if a.payment_method == CASH) \
-                                 + sum(s.cash for s in sales if s.payment_method == CASH)
+            params['total_cash'] = sum(a.amount_paid for a in attendees if a.payment_method == c.CASH) \
+                                 + sum(s.cash for s in sales if s.payment_method == c.CASH)
             params['total_credit'] = sum(a.amount_paid for a in attendees if a.payment_method in [c.STRIPE, c.SQUARE, c.MANUAL]) \
                                    + sum(s.cash for s in sales if s.payment_method == c.CREDIT)
         else:
             params['endday'] = localized_now().strftime('%Y-%m-%d')
             params['endhour'] = localized_now().strftime('%H')
             params['endminute'] = localized_now().strftime('%M')
-
-        stations = sorted(filter(bool, Attendee.objects.values_list('reg_station', flat=True).distinct()))
+        # list all reg stations associated with attendees and sales
+        stations_attendees = session.query(Attendee.reg_station).filter(Attendee.reg_station is not None, Attendee.reg_station > 0)
+        stations_sales = session.query(Sale.reg_station).filter(Sale.reg_station is not None, Sale.reg_station > 0)
+        stations = sorted([r for (r,) in stations_attendees.union(stations_sales).distinct()])
```

Comment:
This is okay as-is, but it may be slightly more efficient to sort the results in the query using the SQLAlchemy `.order_by()` method.

I think it would look like:

```python
stations = [r for (r,) in stations_attendees.union(stations_sales).distinct().order_by('reg_station')]
```

Reply:
With this query..
```
stations_attendees.union(stations_sales).distinct().order_by('reg_station')
```
I'm getting this error:
```
column "reg_station" does not exist
```
This SQL is generated:
```
SELECT DISTINCT anon_1.attendee_reg_station AS anon_1_attendee_reg_station 
  FROM (
       SELECT attendee.reg_station AS attendee_reg_station 
         FROM attendee 
         WHERE attendee.reg_station IS NOT NULL AND attendee.reg_station > 0
       UNION
       SELECT sale.reg_station AS sale_reg_station 
         FROM sale 
         WHERE sale.reg_station IS NOT NULL AND sale.reg_station > 0
       ) AS anon_1
  ORDER BY reg_station
```
I'm having trouble figuring out how to get this to generate correctly. Is there a way to change the alias used by the distinct? Or what am I missing here?

Reply:
Whoops! Yeah there's two things to try (I haven't run this locally, so I'm not 100% sure on this):
`.distinct().alias('reg_station').order_by('reg_station')`
or
`.distinct().label('reg_station').order_by('reg_station')`

Going from memory here, so forgive me if either of those don't do the trick.

Reply:
For those two lines, I got these two errors:
```
AttributeError: 'QuerySubclass' object has no attribute 'alias'
```
```
AttributeError: Neither 'Label' object nor 'Comparator' object has an attribute 'order_by'
```

After much trial and error, here is the first thing that worked:
```
stations_attendees.union(stations_sales).distinct().order_by(Attendee.reg_station)
```

Because the distinct clause is selecting the column from the first half of the union, I am ordering by that column as well. It seems to work as expected.

Thoughts?

Reply:
If sqlalchemy isn't complaining then it's probably doing the right thing. I wasn't sure it would because the second query is selecting from the `sale` table. Sorry for the bad info!

Reply:
I've got `reg_station` values `1` and `2` from the first query, `1` and `3` from the second. This does correctly order them asc and desc, so it must understand what I mean when I put in only the first field.

Thanks for working through this with me :)